src/main/reactjs/src/redux/selectors/clerkTranslator.ts

```diff
@@ -36,9 +42,41 @@ export const selectFilteredClerkTranslators = createSelector(
   (state: RootState) => state.clerkTranslator.translators,
   (state: RootState) => state.clerkTranslator.filters,
   (translators, filters) => {
-    return translators.filter((t) =>
-      filterByAuthorisationStatus(t, filters.authorisationStatus)
+    const currentDate = DateUtils.dateAtStartOfDay(new Date());
+
+    let filtered = translators.filter((t) =>
+      filterByAuthorisationStatus(t, filters.authorisationStatus, currentDate)
     );
+
+    if (filters.name) {
+      const nameFilter = filters.name;
+      filtered = filtered.filter((t) => filterByName(t, nameFilter));
+    }
+
+    if (filters.town) {
+      const town = filters.town;
+      filtered = filtered.filter((t) => filterByTown(t, town));
+    }
+
+    if (filters.fromLang) {
+      const fromLang = filters.fromLang;
+      filtered = filtered.filter((t) => filterByFromLang(t, fromLang));
+    }
+
+    if (filters.toLang) {
+      const toLang = filters.toLang;
+      filtered = filtered.filter((t) => filterByToLang(t, toLang));
+    }
+
+    if (filters.authorisationBasis) {
+      const authorisationBasis =
+        filters.authorisationBasis as AuthorisationBasis;
+      filtered = filtered.filter((t) =>
+        filterByAuthorisationBasis(t, authorisationBasis)
+      );
+    }
```

Comment:
Sen sijaan näistä suodatusehdoista en ole lainkaan niin varma - ainakin tämä toteutus tuntuu hieman ongelmalliselta, jos on tarkoitus että listauksessa on nimenomaisesti kyse **kääntäjistä** eikä **auktorisoinneista**. 

Ongelmakohtia tai avoimia kysymyksiä:
1. Mikäli kääntäjällä on useita auktorisointeja, onko OK että listataan per kääntäjärivi myös sellaiset auktorisoinnit jotka eivät täytä hakuehtoja?
2. Pitäisikö näitä auktorisointitietoihin liittyviä hakuehtoja tarkastella yhtenä könttänä per auktorisointi vai yksi kerrallaan? Ts. jos kääntäjä on esimerkiksi auktorisoitu täsmälleen kielipareihin `FI-SV` sekä `FI-EN`, pitäisikö hänet pitää mukana listauksessa mikäli hakukriteereinä on ollut `from: FI` ja `to: EN`? Kysymys pätee toki muihinkin  auktorisointitietoihin kuin kielet, siis myös perusteen sekä voimassaoloajan (erilliset suodatuspainikkeet!) osalta.

Reply:
Ratkaistu dailyssä:
1. On ok.
2. Auktorisointikohtaiset suodattimet yhdistetään: kääntäjällä on oltava jokin auktorisointi joka toteuttaa kaikki hakuehdot.